diff --git a/sql/sql.js b/sql/sql.js
index e04cb9d..42ec6d2 100644
--- a/sql/sql.js
+++ b/sql/sql.js
@@ -168,12 +168,14 @@
 			if (is(chunk, Subquery)) {
 				if (chunk._.isWith) return {
 					sql: escapeName(chunk._.alias),
 					params: []
 				};
+				const columnList = chunk._.columnList ? sql` (${sql.join(chunk._.columnList.map(sql.identifier), new StringChunk(","))})` : void 0;
 				return this.buildQueryFromSourceParams([
 					new StringChunk("("),
 					chunk._.sql,
 					new StringChunk(") "),
-					new Name(chunk._.alias)
+					new Name(chunk._.alias),
+					columnList
 				], config);
 			}
diff --git a/subquery.d.ts b/subquery.d.ts
index ffcf891..9a4992b 100644
--- a/subquery.d.ts
+++ b/subquery.d.ts
@@ -12,8 +12,9 @@ declare class Subquery<TAlias extends string = string, TSelectedFields extends R
     alias: TAlias;
     isWith: boolean;
     usedTables?: string[];
+    columnList: string[] | undefined;
   };
-  constructor(sql: SQL, fields: TSelectedFields, alias: string, isWith?: boolean, usedTables?: string[]);
+  constructor(sql: SQL, fields: TSelectedFields, alias: string, isWith?: boolean, usedTables?: string[], columnList?: string[]);
 }
 declare class WithSubquery<TAlias extends string = string, TSelection extends Record<string, unknown> = Record<string, unknown>> extends Subquery<TAlias, TSelection> {
   static readonly [entityKind]: string;
diff --git a/subquery.js b/subquery.js
index ccfaf09..a95cc20 100644
--- a/subquery.js
+++ b/subquery.js
@@ -3,13 +3,14 @@
 //#region src/subquery.ts
 var Subquery = class {
 	static [entityKind] = "Subquery";
-	constructor(sql, fields, alias, isWith = false, usedTables = []) {
+	constructor(sql, fields, alias, isWith = false, usedTables = [], columnList) {
 		this._ = {
 			brand: "Subquery",
 			sql,
 			selectedFields: fields,
 			alias,
 			isWith,
-			usedTables
+			usedTables,
+			columnList
 		};
 	}
